Decode/PerfItemValue.cs

```diff
@@ -629,83 +654,250 @@ public string GetString()
                     Value128Append(sb, 0);
                     break;
                 case EventHeaderFieldEncoding.ZStringChar8:
+                    this.StringAppend(sb, convertOptions, Text.Encoding.UTF8);
+                    break;
+                case EventHeaderFieldEncoding.ZStringChar16:
+                case EventHeaderFieldEncoding.StringLength16Char16:
+                    this.StringAppend(
+                        sb,
+                        convertOptions,
+                        this.ByteReader.FromBigEndian ? Text.Encoding.BigEndianUnicode : Text.Encoding.Unicode);
+                    break;
+                case EventHeaderFieldEncoding.ZStringChar32:
+                case EventHeaderFieldEncoding.StringLength16Char32:
+                    this.StringAppend(
+                        sb,
+                        convertOptions,
+                        this.ByteReader.FromBigEndian ? PerfConvert.EncodingUTF32BE : Text.Encoding.UTF32);
+                    break;
+                case EventHeaderFieldEncoding.BinaryLength16Char8:
                 case EventHeaderFieldEncoding.StringLength16Char8:
                     switch (this.Metadata.Format)
                     {
-                        case EventHeaderFieldFormat.HexBytes:
-                            PerfConvert.HexBytesAppend(sb, this.Bytes);
+                        default:
+                        case EventHeaderFieldFormat.Default:
+                        EventHeaderFieldFormatDefault:
+                            if (baseEncoding == EventHeaderFieldEncoding.BinaryLength16Char8)
+                            {
+                                goto case EventHeaderFieldFormat.HexBytes;
+                            }
+                            else
+                            {
+                                goto case EventHeaderFieldFormat.StringUtf;
+                            }
+                        case EventHeaderFieldFormat.UnsignedInt:
```

Comment:
Note: New semantics for Variable-length-blob-encoding + fixed-length-format -- treat as "nullable value".